src/api-next/api-next.ts

```diff
@@ -2189,4 +2187,37 @@ export class IBApiNext {
       }
     );
   }
+
+  /** symbolSamples event handler. */
+  private readonly onSymbolSamples = (
+    subscriptions: Map<number, IBApiNextSubscription<ContractDescription[]>>,
+    reqId: number,
+    contractDescriptions: ContractDescription[]
+  ): void => {
+    const sub = subscriptions.get(reqId);
+    subscriptions.delete(reqId);
+    sub?.next({
+      all: contractDescriptions,
+    });
+    sub?.complete();
+  };
+
+  /**
+   * Search contracts there name of symbol matches the given text pattern.
+   *
+   * @param pattern Either start of ticker symbol or (for larger strings) company name.
+   */
+  searchContracts(pattern: string): Promise<ContractDescription[]> {
```

Comment:
Love the new feature. 

Thinking of 2 additional params I would use on my search.

`Type SymbolSearchOptions {
  exactMatch: boolean,
  secType: SecType
}`
1) Security Type - Filters for only the particular security type.
2) Symbol Exact match boolean, or default to `true` and enable the option for searching with little bit looser ticker input

Reply:
In order to accomplish the Exact match, contractDescription ticker needs to be filtered by search string.

Reply:
Exact match on what? If I do searchContracts("Apple"), there is no exact match as name is not part of result and symbol is AAPL.
It's actually pretty easy to afterwards, at user sider, like:
````
.searchContracts("Apple")
      .then((searchResult) => {
        // filter for AAPL symbol
        searchResult = searchResult.filter((v) => v.contract.symbol == "AAPL");
        // filter for SecType.OPT on derivativeSecTypes 
        searchResult = searchResult.filter(
          (v) =>
            v.derivativeSecTypes?.find((v) => v === SecType.OPT) !== undefined
        );
        ...
```